### ISSUE TITLE
Rename refactoring does not trigger build

### ISSUE THREAD

**Issue by lsotnk** [eclipse/xtext-eclipse#1029]
Hi.

Even though the following example is written in Xtend, I'm pretty sure it might be a general Xtext-problem, since I've noticed a similar problem with my own language: 

1. Create file Foo.xtend:
```
package bug
class Foo {
	public String name
}
```
2. Create file Bar.xtend:

```
package bug
class Bar {
	def static void main(String[] args) {
		val foo = new Foo
		println(foo.name)		
	}
}
```
3. Open Xtext Build Console
4. Rename field _name_ to _name2_ in class _Foo_. You will notice the Xtext Build Console does not log anything
5. Open Bar.xtend
6. Modify the file in the editor (e.g. insert a line feed somewhere). _foo.name2_ is now marked with an error marker.

**Comment by cdietrich** [comment]
refactoring should alread rename name in file bar.
=> i am not sure if i can follow you

**Comment by cdietrich** [comment]
@lsotnk  are you sure this is about build and not dirty editors?

**Comment by cdietrich** [comment]
analysis: there is no build.
the java file gets refactored 
thus no error message.

**Comment by lsotnk** [comment]
@cdietrich Yes, I mean the build. I suppose the Xtext index wasn't updated, since there was no build. And even though everything was correctly renamed foo.name2 couldn't be found.
If I modify the workspace somehow after the refactoring (e.g. create a new file), Foo.xtend is built together with the other changes and foo.name2 can be resolved successfully.

**Comment by cdietrich** [comment]
what is even more wired: when using the rename dialog (little arrow)
then it seems to work

**Comment by cdietrich** [comment]
=> i wonder if this is a bug in ltk or how we use it

**Comment by cdietrich** [comment]
@szarnekow any idea? i am rather clueless

**Comment by cdietrich** [comment]
ping @szarnekow

**Comment by cdietrich** [comment]
ping @szarnekow

**Comment by lsotnk** [comment]
I think the problem might have been introduced with https://github.com/eclipse/xtext-eclipse/commit/f48945de5eb1ffaec688599267627b6626ea86e5.

syncUtil.waitForBuild is just a call to IWorkspace.build. If the line is inserted again after syncUtil.reconcileAllEditors the build runs automatically after the refactoring.

**Comment by cdietrich** [comment]
move to 2.24
